x-pack/plugin/sql/qa/server/src/main/resources/pivot.csv-spec

```diff
@@ -201,6 +201,29 @@ null                 |10044          |Mingsen        |F              |1994-05-21
 // end::sumWithoutSubquery
 ;
 
+sumWithInnerAggregateSumOfSquares
+schema::birth_date:ts|emp_no:i|first_name:s|gender:s|hire_date:ts|last_name:s|1:d|2:d
+SELECT * FROM test_emp PIVOT (SUM_OF_SQUARES(salary) FOR languages IN (1, 2)) LIMIT 5;
```

Comment:
I'd like to see a test with multiple arguments that get folded into compound aggs, such as sum / min / max.

Reply:
Do you mean multiple aggregations inside the PIVOT? Currently there is a check that we can only have a single aggregation inside the `PIVOT`, so we can't have `PIVOT(MIN(salary), MAX(salary) FOR ...)`.

This change only lifts the limitation on the *type* of that single aggregation within the PIVOT, so aggregations that are translated into an `InnerAggregate` will be allowed.

Reply:
Is there any validation or check for this (now that PostOptimizerTest has been removed)? this goes beyond the scope of this ticket so please raise a ticket if this functionality is missing or needs to be further investigated.

Reply:
The `PostOptimizerTests` checked that the *type* of the aggregation cannot be one that results in an InnerAggregate vs the number of the aggregations inside the PIVOT.

There is already a check in the `LogicalPlanBuilder` that only allows one aggregation inside the PIVOT: https://github.com/elastic/elasticsearch/blob/2f5ab81b9ad884434d1faa60f4391f966ede73e8/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/LogicalPlanBuilder.java#L186

I cannot see it tested in unit tests, but it is there and works. Added: #65894